dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,969 @@
+                        .append("Categories=$contents.getEmptyList())");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("CategoriesNames=$contents.getEmptyList())");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("=$contents.getEmptyList())");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("CategoriesKeys=$contents.getEmptyList())");
+                }
+            } else if (field instanceof TagField) {
+                content.setTags();
+                String value = content.getStringProperty(field.variable());
+                sb.append("#set($")
+                    .append(field.variable())
+                    .append("=\"")
+                    .append(UtilMethods.espaceForVelocity(value)
+                        .trim())
+                    .append("\")");
+            }
+        }
+
+
+        // get the contentlet categories to make a list
+        String categories = "";
+        Set<Category> categoryList = new HashSet<Category>(categoryAPI.getParents(content, systemUser, false));
+        if (categoryList != null && categoryList.size() > 0) {
+            StringBuilder catbuilder = new StringBuilder();
+            Iterator<Category> it = categoryList.iterator();
+            while (it.hasNext()) {
+                Category category = (Category) it.next();
+                catbuilder.append("\"")
+                    .append(category.getInode())
+                    .append("\"");
+                if (it.hasNext()) {
+                    catbuilder.append(",");
+                }
+            }
+            categories = catbuilder.toString();
+
+            sb.append("#set($ContentletFilteredCategories=$categories.filterCategoriesByUserPermissions([")
+                .append(categories)
+                .append("] ))");
+            sb.append("#set($ContentletCategories=$categories.fetchCategoriesInodes($ContentletFilteredCategories))");
+            sb.append("#set($ContentletCategoryNames=$categories.fetchCategoriesNames($ContentletFilteredCategories))");
+            sb.append("#set($ContentletCategoryKeys=$categories.fetchCategoriesKeys($ContentletFilteredCategories))");
+        } else {
+            sb.append("#set($ContentletFilteredCategories=$contents.getEmptyList())");
+            sb.append("#set($ContentletCategories=$contents.getEmptyList())");
+            sb.append("#set($ContentletCategoryNames=$contents.getEmptyList())");
+            sb.append("#set($ContentletCategoryKeys=$contents.getEmptyList())");
+        }
+
+        // This needs to be here because the all fields like cats etc.. need to be parsed first and
+        // it needs to be before
+        // the $CONTENT_INODE is reset sb.append("#set( $CONTENT_INODE=\"" + content.getInode() +
+        // "\" )");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)